### ISSUE TITLE
Can't import universal numpy on MacOS Snow Leopard when running 32 bit python (migrated from Trac #1221)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2774]
Original ticket http://projects.scipy.org/numpy/ticket/1221
Reported 2009-09-13 by trac user rwest, assigned to unknown.

After building 32/64-bit numpy from SVN on MacOS 10.6 Snow Leopard according to the instructions at 
http://blog.hyperjeff.net/?p=160
I would expect to have a universal (i386 and x86_64) version of numpy.

If, however, I start a 32-bit python shell and import numpy then I get a "ValueError: array is too big." on line line 600 of numpy/core/numerictypes.py:

---

These are my steps:

Installing numpy:

```
sudo mv /System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/numpy \
/System/Library/Frameworks/Python.framework/Versions/2.6/Extras/lib/python/numpy_APPLE_DEFAULT
export MACOSX_DEPLOYMENT_TARGET=10.6
export CFLAGS="-arch i386 -arch x86_64"
export FFLAGS="-arch i386 -arch x86_64"
export LDFLAGS="-Wall -undefined dynamic_lookup -bundle -arch i386 -arch x86_64"
cd ~/tmp
svn co http://svn.scipy.org/svn/numpy/trunk numpy
cd numpy
python setup.py build --fcompiler=gnu95
sudo python setup.py install
```

Loading 32-bit python:
    export VERSIONER_PYTHON_PREFER_32_BIT=yes 
    python 

Trying to import numpy:
    Python 2.6.1 (r261:67515, Jul  7 2009, 23:51:51) 
    [GCC 4.2.1 (Apple Inc. build 5646)] on darwin
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import numpy
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/Library/Python/2.6/site-packages/numpy/**init**.py", line 130, in <module>
        import add_newdocs
      File "/Library/Python/2.6/site-packages/numpy/add_newdocs.py", line 9, in <module>
        from lib import add_newdoc
      File "/Library/Python/2.6/site-packages/numpy/lib/**init**.py", line 4, in <module>
        from type_check import *
      File "/Library/Python/2.6/site-packages/numpy/lib/type_check.py", line 8, in <module>
        import numpy.core.numeric as _nx
      File "/Library/Python/2.6/site-packages/numpy/core/__init__.py", line 8, in <module>
        import numerictypes as nt
      File "/Library/Python/2.6/site-packages/numpy/core/numerictypes.py", line 600, in <module>
        _typestr[key] = empty((1,),key).dtype.str[1:]
    ValueError: array is too big.

This is mentioned at http://mail.scipy.org/pipermail/scipy-user/2009-May/021090.html
but I have not found a solution.

**Comment by thouis** [comment]
Comment in Trac by trac user jbsnyder, 2009-09-15

I can confirm this issue as well.

Interestingly, an install of NumPy 1.2.1, which is bundled with Apple's build of Python 2.6.1, provided with Snow Leopard does not have this problem.

**Comment by thouis** [comment]
Comment in Trac by trac user jbsnyder, 2009-09-15

**Comment by thouis** [comment]
Comment in Trac by trac user guidon, 2009-10-22

Hi. I was wondering if you found a solution to this problem. I have the same problem (below). 

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/**init**.py", line 130, in <module>
    import add_newdocs
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/add_newdocs.py", line 9, in <module>
    from lib import add_newdoc
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/lib/**init**.py", line 4, in <module>
    from type_check import *
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/lib/type_check.py", line 8, in <module>
    import numpy.core.numeric as _nx
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/core/__init__.py", line 8, in <module>
    import numerictypes as nt
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/numpy-1.4.0.dev7542-py2.6-macosx-10.6-universal.egg/numpy/core/numerictypes.py", line 737, in <module>
    _typestr[key] = empty((1,),key).dtype.str[1:]
ValueError: array is too big.

Any tips to solve this problem?

Thanks,
- Guido

**Comment by thouis** [comment]
Comment in Trac by trac user witzclassic, 2009-11-01

Guido, I have the same problem.   It seems that 2.6.1 is still in /usr/bin/python and works OK.  2.6.1 installed with snow leopard supports x86_64 whereas 2.6.4 seems not to.  I don't know if that is a problem with my install or the disk image itself but reverting to 2.6.1 seems to work around the issue.  See below and/or import platform.architecture() which shows 32 v. 64.

"file /usr/bin/python": (2.6.1 that came with snow leopard)
usr/bin/python: Mach-O universal binary with 3 architectures
/usr/bin/python (for architecture x86_64):  Mach-O 64-bit executable x86_64
/usr/bin/python (for architecture i386):    Mach-O executable i386
/usr/bin/python (for architecture ppc7400): Mach-O executable ppc

file python2.6: (2.6.4 that I installed from the web site):
python2.6: Mach-O universal binary with 2 architectures
python2.6 (for architecture ppc):       Mach-O executable ppc
python2.6 (for architecture i386):      Mach-O executable i386
- Steve

**Comment by thouis** [comment]
Comment in Trac by atmention:robince, 2009-11-03

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-11-25

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-11-27

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-11-27

The problem is that our headers do no support universal builds. We have to special case the macro defined by the config stage for mac os x and hardcode them.

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-11-27

Should be fixed in r7793 in the trunk (backported to 1.4.x branch as well for 1.4.0).
